addons/hr_timesheet/__init__.py

```diff
@@ -48,3 +48,8 @@ def _uninstall_hook(cr, registry):
     project_ids = env['res.company'].search([('internal_project_id', '!=', False)]).mapped('internal_project_id')
     if project_ids:
         project_ids.write({'active': False})
+
+    stage_copy = env.ref('hr_timesheet.internal_project_default_stage').copy()
+    env['project.task'].with_context(active_test=False).search(
+        [('stage_id', '=', env.ref('hr_timesheet.internal_project_default_stage').id)]
+    ).stage_id = stage_copy
```

Comment:
Copying looks like a bad idea to me.
Why don't you simply delete the corresponding `ir.model.data` record?  When deleted, the existing stage remains, but is no longer "recognized" as `hr_timesheet.internal_project_default_stage`...

Reply:
Oh, that's pretty cool! Thank you :)